cvmfs/mountpoint.cc

```diff
@@ -1455,6 +1455,53 @@ bool MountPoint::CreateDownloadManagers() {
     download_mgr_->ShardProxies();
   }
 
+  // configure http tracing header
+  if (options_mgr_->GetValue("CVMFS_HTTP_TRACING", &optarg) &&
+      options_mgr_->IsOn(optarg)) {
+    download_mgr_->EnableHTTPTracing();
+    if (options_mgr_->GetValue("CVMFS_HTTP_TRACING_HEADERS", &optarg)) {
+      std::vector<std::string> tokens = SplitString(optarg, '|');
+      sanitizer::AlphaNumSanitizer sanitizer;
+
+      for (unsigned int i = 0; i < tokens.size(); i++) {
+        std::string token = Trim(tokens[i]);
+
+        std::vector<std::string> key_val = SplitString(token, ':');
+
+        if (key_val.size() != 2) {
+          LogCvmfs(kLogCvmfs, kLogSyslog | kLogDebug,
+             "Http tracing header: Skipping current token part of "
+             "CVMFS_HTTP_TRACING_HEADERS! Invalid "
+             "<key:value> pair. Token: %s", token.c_str());
+          continue;
+        }
+
+        std::string prefix = "X-CVMFS-";
+        std::string key = Trim(key_val[0]);
+
+        if (!sanitizer.IsValid(key)) {
+          // check if the key starts already with the X-CVMFS- prefix and that
+          // no other "-" are part of it. If yes accept it as valid
+          // tracing header
+          if (key.compare(0, prefix.size(), prefix) == 0 &&
+              SplitString(key, '-').size() == 3) {
+            prefix = "";
```

Comment:
Should we go through the hassle to support this? We could simply require headers to be specified without the X-CVMFS- part

Reply:
decision was to remove it for the moment